hibernate-core/src/test/java/org/hibernate/orm/test/filter/FilterDefinitionOrderTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.orm.test.filter;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import org.hibernate.annotations.Filter;
+import org.hibernate.annotations.FilterDef;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.JiraKey;
+import org.junit.jupiter.api.Test;
+
+@DomainModel(
+        annotatedClasses = {
+                FilterDefinitionOrderTest.AMyEntity.class,
+                FilterDefinitionOrderTest.XEntity.class
+        }
+)
+@JiraKey("HHH-19036")
+public class FilterDefinitionOrderTest extends AbstractStatefulStatelessFilterTest {
+
+    @Test
+    public void smokeTest() {
+        // nothing really to test here,
+        // if the test starts, then everything is great!
```

Comment:
Might as well test that enabling the filter works, and that the default condition is actually applied while we're at it.

Reply:
so something like this ? 😃